synapse/tools/genpkg.py

```diff
@@ -55,6 +56,34 @@ def loadPkgProto(path, opticdir=None):
     protodir = os.path.dirname(full)
     pkgname = pkgdef.get('name')
 
+    logodef = pkgdef.get('logo')
+    if logodef is not None:
+
+        path = logodef.pop('path', None)
+
+        if path is not None:
+            with s_common.reqfile(protodir, path) as fd:
+                logodef['file'] = base64.b64encode(fd.read()).decode()
+
+        if logodef.get('mime') is None:
+            mesg = 'Mime type must be specified for logo file.'
+            raise s_exc.BadPkgDef(mesg=mesg)
+
+    for docdef in pkgdef.get('docs', ()):
+
+        if docdef.get('title') is None:
+            mesg = 'Each entry in docs must have a title.'
+            raise s_exc.BadPkgDef(mesg=mesg)
+
+        path = docdef.pop('path', None)
+        if path is not None:
+            with s_common.reqfile(protodir, path) as fd:
+                docdef['content'] = fd.read().decode()
+
+        if docdef.get('content') is None:
+            mesg = 'Each entry in docs must have a content valu.'
```

Comment:
Might be good to have this error mesg specify that the 'content' value can come from a file specified with the 'path' key